### PR TITLE
feat: goroutine leak check

### PR DESCRIPTION
**Pull request by istae**
This is combined PR for all changes related to the issue #3516

Closes:  #3522

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3598)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

**Review by janos — APPROVED** [review]

File: pkg/api/pss.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func (s *Service) pumpWs(conn *websocket.Conn, t string) {
 	defer s.wsWg.Done()
 
 	var (
-		dataC  = make(chan []byte)
+		dataC  = make(chan []byte, 32)
```
Comment (vladopajic) [inline pkg/api/pss.go:152]: remove buffer from dataC channel

File: pkg/feeds/sequence/sequence.go [hunk 2]
```diff
@@ -233,12 +233,13 @@ func (f *asyncFinder) at(ctx context.Context, at int64, min int, i *interval, c
 			}
```
Comment (vladopajic) [inline pkg/feeds/sequence/sequence.go:233]: add select on `quit` channel

File: pkg/feeds/sequence/sequence.go [hunk 3]
```diff
@@ -158,7 +158,7 @@ func (f *asyncFinder) At(ctx context.Context, at, after int64) (ch swarm.Chunk,
 		return nil, nil, &index{uint64(after)}, nil
 	}
 	// if chunk exists construct an initial interval with base=0
-	c := make(chan *result)
+	c := make(chan *result, 8)
```
Comment (vladopajic) [inline pkg/feeds/sequence/sequence.go:161]: remove buffer

File: pkg/p2p/libp2p/internal/reacher/reacher.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ func (r *reacher) manage() {
 
 	defer r.wg.Done()
 
-	c := make(chan *peer)
+	c := make(chan *peer, 32)
```
Comment (vladopajic) [inline pkg/p2p/libp2p/internal/reacher/reacher.go:95]: remove buffer

File: pkg/p2p/p2p.go [hunk 5]
```diff
@@ -114,6 +114,7 @@ type ReachableNotifier interface {
 type Reacher interface {
 	Connected(swarm.Address, ma.Multiaddr)
 	Disconnected(swarm.Address)
+	Close() error
```
Comment (vladopajic) [inline pkg/p2p/p2p.go:117]: remove this

File: pkg/pusher/pusher.go [hunk 6]
```diff
@@ -107,7 +107,7 @@ func (s *Service) chunksWorker(warmupTime time.Duration, tracer *tracing.Tracer)
 		loggerV1          = logger.V(1).Build()
 		timer             = time.NewTimer(traceDuration)
 		sem               = make(chan struct{}, concurrentPushes)
-		cc                = make(chan *Op)
+		cc                = make(chan *Op, 16)
```
Comment (vladopajic) [inline pkg/pusher/pusher.go:110]: remove buffer

File: pkg/retrieval/retrieval.go [hunk 7]
```diff
@@ -145,7 +145,7 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr, sourcePeerAddr swarm.
 		defer close(done)
 
 		resultC := make(chan retrievalResult, 1)
-		retryC := make(chan struct{})
+		retryC := make(chan struct{}, 32)
```
Comment (vladopajic) [inline pkg/retrieval/retrieval.go:148]: remove buffer

File: pkg/pusher/pusher.go [hunk 8]
```diff
@@ -202,9 +202,14 @@ func (s *Service) chunksWorker(warmupTime time.Duration, tracer *tracing.Tracer)
 				}
 			case apiC := <-s.smuggler:
 				go func() {
-					for op := range apiC {
```
Comment (istae) [inline pkg/pusher/pusher.go:205]: this is fine, we don't need to touch this one.

Comment (vladopajic) [reply pkg/pusher/pusher.go:205]: just double checked it and it looks like change from PR is needed.

tests fail with this error without it:
```
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 13 in state chan receive, with github.com/ethersphere/bee/pkg/pusher.(*Service).chunksWorker.func5.1 on top of the stack:
goroutine 13 [chan receive]:
github.com/ethersphere/bee/pkg/pusher.(*Service).chunksWorker.func5.1()
        /home/vlado/Workspace/bee/pkg/pusher/pusher.go:205 +0x5b
created by github.com/ethersphere/bee/pkg/pusher.(*Service).chunksWorker.func5
        /home/vlado/Workspace/bee/pkg/pusher/pusher.go:204 +0x28a
```

**Comment by vladopajic** [comment]
Closing this PR in favor of #3804

**Pull request closed without merging**